File: Frontend/src/types/contracts/analytics.ts
import { z } from "zod";

export const topicMetricSchema = z.object({
  topic: z.string(),
  score: z.number(),
  trend: z.number(),
  relevance: z.enum(["critical", "high", "normal"]),
});

export const analyticsMicroMetricSchema = z.object({
  key: z.string(),
  label: z.string(),
  value: z.number(),
  delta: z.string(),
  trend: z.array(z.number()),
});

export const analyticsRecentSessionSchema = z.object({
  reportId: z.string(),
  company: z.string(),
  mode: z.string(),
  score: z.number(),
  completedAt: z.string(),
});

export const analyticsOverviewSchema = z.object({
  overallScore: z.number(),
  readinessScore: z.number(),
  streakDays: z.number(),
  improvementPercent: z.number(),
  scoreTrend: z.array(z.number()),
  readinessTrend: z.array(z.number()),
  microMetrics: z.array(analyticsMicroMetricSchema),
  topicPerformance: z.array(topicMetricSchema),
  recentSessions: z.array(analyticsRecentSessionSchema),
});

File: Backend/app/schemas/analytics.py
from app.core.serialization import CamelModel
from app.core.timeutils import UtcDatetime
from app.schemas.common import TopicRelevance


class TopicMetric(CamelModel):
    topic: str
    score: int
    trend: int
    relevance: TopicRelevance


class AnalyticsMicroMetric(CamelModel):
    key: str
    label: str
    value: float
    delta: str
    trend: list[int]


class AnalyticsRecentSession(CamelModel):
    report_id: str
    company: str
    mode: str
    score: int
    completed_at: UtcDatetime


class AnalyticsOverview(CamelModel):
    overall_score: int
    readiness_score: int
    streak_days: int
    improvement_percent: int
    score_trend: list[int]
    readiness_trend: list[int]
    micro_metrics: list[AnalyticsMicroMetric]
    topic_performance: list[TopicMetric]
    recent_sessions: list[AnalyticsRecentSession]
